tests/datawidgetstests/test_autovizwidget.py

```diff
@@ -97,4 +102,5 @@ def test_convert_to_displayable_dataframe():
     assert_equals(result.dtypes[u'bool_col'], object)
     assert_equals(len(result[u'bool_col']), 2)
     assert_equals(result[u'bool_col'][0], 'True')
-    assert_equals(result[u'bool_col'][1], 'False')
\ No newline at end of file
+    assert_equals(result[u'bool_col'][1], 'False')
+    spark_events.emit_graph_render_event.assert_called_once_with(encoding.chart_type)
```

Comment:
The emission happens in `on_render_viz`, so that's a given as long as that method is called whenever the encoding changes.

I don't know if I know enough about the autoviz codebase to simulate the endcoding widget being changed in a unit test, but I guess I could add that if you think it's useful